unified-installer/astra-unified-installer.sh

```diff
@@ -2656,12 +2661,17 @@ step_apply_resources() {
 
         output="$(kubectl apply -k "$operators_dir" 2> "$__ERR_FILE")"
         captured_err="$(get_captured_err)"
-        if echo "$captured_err" | grep -q "Warning:"; then
-            logdebug "captured warning when applying kustomize resources:${__NEWLINE}$captured_err"
-        elif echo "$captured_err" | grep -q "no objects passed to apply"; then
-            logdebug "no kustomize resources to apply, skipping"
-        elif [ -z "$output" ] || [ -n "$captured_err" ]; then
-            add_problem "Failed to apply kustomize resources: $captured_err"
+
+        if [ -n "$captured_err" ]; then
+            while IFS= read -r line; do
+                if echo "$line" | grep -q "Warning:"; then
+                    logdebug "captured warning when applying kustomize resources:${__NEWLINE}$captured_err"
+                elif echo "$line" | grep -q "no objects passed to apply"; then
+                    logdebug "no kustomize resources to apply, skipping"
+                elif [ -z "$output" ] || [ -n "$line" ]; then
+                    add_problem "Failed to apply kustomize resources: $line"
+                fi
+            done < <(echo "$captured_err")
```

Comment:
If the `apply` command has both a set of warnings and errors, the warning would basically "hide" the actual errors since we were checking the whole string at once instead of line by line. We'd then continue with the rest of the script even though the apply failed.

Reply:
Another good catch